### PR TITLE
Finalize top-level API

### PR DESCRIPTION
**Pull request by j-bennet**
- [x] Finalize API for writing to deltalake
- [x] Rename `read_delta_table` -> `read_deltalake`
- [x] Remove `vacuum` and `history`; user can use `deltalake.DeltaTable` for that
- [x] Add documentation for `to_deltalake` to README

Closes https://github.com/dask-contrib/dask-deltatable/issues/42, https://github.com/dask-contrib/dask-deltatable/issues/17, https://github.com/dask-contrib/dask-deltatable/issues/16.

### CODE REVIEW

**Comment by j-bennet** [comment]
Windows tests fail uploading coverage to codecov.

File: dask_deltatable/core.py [hunk 1]
```diff
@@ -319,72 +248,12 @@ def read_delta_table(
...
-    dtw = DeltaTableWrapper(
-        path=path,
-        version=None,
-        columns=None,
-        storage_options=storage_options,
-        delta_storage_options=delta_storage_options,
+def read_delta_table(*args, **kwargs):
+    warnings.warn(
+        "`read_delta_table` was renamed, use `read_deltalake` instead",
+        category=DeprecationWarning,
```
Comment (fjetter) [inline dask_deltatable/core.py:254]: I think this is not necessary. Just remove this. The library hasn't been out for long enough for us to care about this. The last release was more than two years ago

File: README.md [hunk 2]
```diff
@@ -21,9 +21,6 @@ pip install dask-deltatable
    - Parquet filters
      - row filter
      - partition filter
-5. Query Delta commit info and history
-6. API to ``vacuum`` the old / unused parquet files
-7. Load different versions of data by timestamp or version.
```
Comment (fjetter) [inline README.md:26]: This is still true, isn't it?

File: dask_deltatable/core.py (hunk 1 above)
Comment (j-bennet) [reply dask_deltatable/core.py:254]: Ok.

**Review by fjetter — APPROVED** [review]

**Pull request merged**